### PR TITLE
[ILM] Fix race condition in test

### PR DESCRIPTION
**Pull request by AthenaEryma**
Previously, testRunStateChangePolicyWithNextStep asserted that the
ClusterState before and after running the steps were equal. The test
only passed due to a race condition: The latch would be triggered by the
step execution, but the cluster state update thread would continue
running the next stepbefore committing the change to the cluster state. 
This allowed the test to read the old cluster state and pass the equality 
check about 99.99% of the time.

The test now waits for the new cluster state to be committed by
checking that it is _not_ equal to the old cluster state.

This should resolve this flaky test issue.

### CODE REVIEW

**Review by talevy — APPROVED** [review]

**Comment by gwbrown** [comment]
@elasticmachine run sample packaging tests

This appears to be the issue that made every build fail for a while.

**Comment by talevy** [comment]
yup. the fix was commited, just need to wait for the CI nodes to pick up latest

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunnerTests.java [hunk 1]
```diff
@@ -259,14 +259,15 @@ public void testRunStateChangePolicyWithNextStep() throws Exception {
...
 
-        latch.await(5, TimeUnit.SECONDS);
-        ClusterState after = clusterService.state();
+        assertTrue(latch.await(5, TimeUnit.SECONDS));
 
-        assertEquals(before, after);
+        // The cluster state can take a few extra milliseconds to update after the steps are executed
+        assertBusy(() -> assertNotEquals(before, clusterService.state()));
         assertThat(step.getExecuteCount(), equalTo(1L));
+        assertThat(nextStep.getExecuteCount(), equalTo(1L));
```
Comment (colings86) [inline x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunnerTests.java]: I think we should also check that the clusterstate shows the correct ILM state at this point too

**Review by colings86 — CHANGES_REQUESTED** [review]

File: x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunnerTests.java (hunk 1 above)
Comment (gwbrown) [reply x-pack/plugin/ilm/src/test/java/org/elasticsearch/xpack/indexlifecycle/IndexLifecycleRunnerTests.java]: Done.  There's a few other tests in this class that could probably use the same treatment, but that's a separate PR.

**Review by colings86 — APPROVED** [review]
LGTM

**Pull request merged**